python-package/lightgbm/dask.py

```diff
@@ -176,12 +177,74 @@ def _train_part(
     else:
         group = None
 
+    # construct local eval_set data.
+    local_eval_set, local_eval_sample_weight, local_eval_group = None, None, None
+    n_evals = max([len(x.get('eval_set', [])) for x in list_of_parts])
+    has_eval_weights = any([x.get('eval_sample_weight') is not None for x in list_of_parts])
+    if n_evals:
+
+        local_eval_set = []
+        if has_eval_weights:
+            local_eval_sample_weight = list()
+        if is_ranker:
+            local_eval_group = list()
+
+        # consolidate parts of each individual eval component.
+        for i in range(n_evals):
+            x_e, y_e, w_e, g_e = [], [], [], []
```

Comment:
```suggestion
            x_e = []
            y_e = []
            w_e = []
            g_e = []
```

Same comment as above. I have a preference for one assignment per line